Sources/App/Core/MarkdownHTMLConverter.swift

```diff
@@ -0,0 +1,80 @@
+import libcmark_gfm
+
+struct MarkdownHTMLConverter {
+    
+    enum ConversionError: Error {
+        case failedToCreateParser
+        case markdownToASTError
+        case astRenderError
+        case htmlToStringFailed
+    }
+    
+    let markdown: String
+    let enableGFM: Bool
+    
+    let options = CMARK_OPT_SAFE
+    
+    private func createCommonMarkNode(markdown: String) throws -> UnsafeMutablePointer<cmark_node> {
+        let tree = markdown.withCString { (document: UnsafePointer<Int8>) -> UnsafeMutablePointer<cmark_node>? in
+            let stringLength = Int(strlen(document))
+            return cmark_parse_document(document, stringLength, options)
+        }
+        
+        guard let ast = tree else {
+            throw ConversionError.markdownToASTError
+        }
+        
+        return ast
+    }
+    
+    private func createGFMCommonMarkNode(markdown: String) throws -> UnsafeMutablePointer<cmark_node> {
+        cmark_gfm_core_extensions_ensure_registered()
```

Comment:
Do we have a link for the docs of the C-interface? Would be good to link to I think as the initialisation steps aren't really all that obvious 😆

Reply:
Sorry docs? What are these? 😂

I figured this all out by basically educated guessing and combing through a couple different repos which used swift-cmark to try and get the right flow. Never found solid docs on it all or a 'simple' one-page solution.